code: add game modes menu with classic and random options

LeticiaBN opened the pull request:
## Description
Adds a new Game Modes selection screen accessible from the main menu. Users can choose between Classic Snake Game mode or Random mode (which randomly selects a game mode when starting). The selected mode is displayed with visual indicators (>> <<) in the game modes menu and shown below the title on the main menu.

## Related Issue
Closes #343

## Changes Made
- Added new `GameModesScene` with Classic and Random mode options
- Added "Game Modes" button to main menu (between Start Game and Settings)
- Implemented visual indicator (>> <<) for currently selected mode
- Display selected game mode below title on main menu
- Random mode infrastructure ready for future game mode additions
- Added mode description when hovering over Random option

## Screenshots
Main Menu with selected mode displayed:
- Selected game mode shown below title in gray text

<img width="832" height="824" alt="Screenshot 2025-11-16 at 10 47 06" src="https://github.com/user-attachments/assets/694e9330-3302-481b-a32d-9bbc163d0594" />


Game Modes Menu:
- Two options: "Classic Snake Game" and "Random"
- Visual indicator (>>) shows confirmed selection
- ESC returns to main menu

<img width="833" height="834" alt="Screenshot 2025-11-16 at 10 51 04" src="https://github.com/user-attachments/assets/230f1d28-ed20-4994-8bd4-f6b001598189" />

LeticiaBN committed:
code: add game modes menu with classic and random options

Implements new game modes selection screen accessible from main menu.
Users can choose between Classic Snake Game and Random mode (which
randomly selects a game mode). Selected mode is displayed with visual
indicators and shown below title on main menu. Enhanced main menu
aesthetics with larger title and improved spacing.

LeticiaBN committed:
minor: adjust random mode description position closer to button

LeticiaBN committed:
fix random name

davigdomingues commented:
@LeticiaBN Although the proposed code, at first glance, fulfills the requirements of the attached issue, the correct reset for the selected game mode is not occurring, since the settings still store (perhaps via localhost) the previously customized settings made by the player in the "settings" option. I ran some tests and when selecting the "Classic Snake Game" mode, the gameplay settings are not "set" to the default gameplay configuration, just as when choosing the "Random" game mode, the game options preserve everything the user configured, as described previously.

davigdomingues commented:
Here are some iterations I've done:

- "Manual" settings:
<img width="646" height="660" alt="image" src="https://github.com/user-attachments/assets/5e0ad877-33d2-4206-a328-247b03502196" />


- Gameplay:
<img width="644" height="674" alt="image" src="https://github.com/user-attachments/assets/a0857046-cc7a-4539-8b34-7a188f6e8521" />


- Menu:
<img width="644" height="665" alt="image" src="https://github.com/user-attachments/assets/bc04c234-9d6e-4da7-b3ee-7124a9dbd392" />


- Result (same configurations as the "Manual Settings):
<img width="641" height="674" alt="image" src="https://github.com/user-attachments/assets/b6641e60-f718-4f19-bc7b-0c04360d3a47" />


- Classic mode choosed:
<img width="639" height="620" alt="image" src="https://github.com/user-attachments/assets/71137308-97de-4550-bfe3-b2588fdb8493" />


- Result (still the same "manual" configurations):
<img width="641" height="677" alt="image" src="https://github.com/user-attachments/assets/c34954f4-549a-41f7-a301-2e5f2edb6884" />

davigdomingues reviewed (changes requested):
Reasons presented in "Conversations" area.

LeticiaBN commented:
> Reasons presented in "Conversations" area.

Thanks for the review! 

@davigdomingues 
I think this behavior is expected, since the settings represent the customizations the player chooses. I think that The Classic Snake Game is defined as the mode where the snake grows when eat apple and apple appear one pixel at a time, and players can use the Settings menu to diversify this gameplay with additional configurations.

Do you think it would be helpful to add a “Reset to default” button in the Settings menu to restore the original gameplay configuration? Or would it be better if selecting Classic Snake Game automatically reset the settings to the default values?

davigdomingues commented:
@LeticiaBN Firstly, your vision of what the classic game mode would be is the same as mine, as well as the alternative gameplay that the user can configure, so I agree with your comment. 

Secondly, I think it would be better if selecting Classic Snake Game automatically reset the settings to the default values. Also, I wouldn't be contrary to the possibility of having the "Reset to default" functionality.

LeticiaBN committed:
code: add reset to default button and auto-reset for classic mode

Add 'Reset to Default' button at the bottom of Settings menu that
restores all settings to default values. When Classic Snake Game mode
is selected, settings automatically reset to defaults. Custom settings
persist when manually changed and can be used with any game mode.

LeticiaBN commented:
> @LeticiaBN Firstly, your vision of what the classic game mode would be is the same as mine, as well as the alternative gameplay that the user can configure, so I agree with your comment.
> 
> Secondly, I think it would be better if selecting Classic Snake Game automatically reset the settings to the default values. Also, I wouldn't be contrary to the possibility of having the "Reset to default" functionality.

I agree with that. I updated the PR, can you take a look? @davigdomingues 

I added the "Reset to default" button and reset when selecting Classic Snake Game

davigdomingues commented:
@LeticiaBN Ok, I'll take a look 👍

davigdomingues commented:
@LeticiaBN I saw the implementation of the reset to the default configuration in the "settings" area and I confirm that the "Classic Snake Game" option is also functional. However, I have only one question: when choosing random mode, will the randomness of the gameplay come from a range of specific game types or from a completely random configuration from "settings," or both?

LeticiaBN commented:
> @LeticiaBN I saw the implementation of the reset to the default configuration in the "settings" area and I confirm that the "Classic Snake Game" option is also functional. However, I have only one question: when choosing random mode, will the randomness of the gameplay come from a range of specific game types or from a completely random configuration from "settings," or both?

@davigdomingues 
We decided in class that the randomness will come from a range of game modes that are described in the next_steps document. It will just select a random game mode and you can change settings to customize.

davigdomingues commented:
@LeticiaBN Thanks for reminding me, that was indeed the decision, lol. Well then, consider your PR approved :)

lfelipediniz reviewed (approved):
Ok!
